components/director/internal/domain/scenarioassignment/resolver.go

```diff
@@ -54,31 +57,22 @@ type Resolver struct {
 	converter     gqlConverter
 	svc           asaService
 	tenantService tenantService
+	fetcher       tenantFetcher
 }
 
 // CreateAutomaticScenarioAssignment missing godoc
 func (r *Resolver) CreateAutomaticScenarioAssignment(ctx context.Context, in graphql.AutomaticScenarioAssignmentSetInput) (*graphql.AutomaticScenarioAssignment, error) {
+	if err := r.fetcher.FetchOnDemand(in.Selector.Value); err != nil {
+		return nil, errors.Wrapf(err, "while trying to create if not exists subaccount %s", in.Selector.Value)
```

Comment:
We should probably add a check here to first fetch the tenant from the Compass DB.
Then, if it does not exist, we "fetch it on demand", and if everything is ok -> continue.

The transaction here is interesting - we can open it in the beginning of the method, if the tenant does not exist -> close it, and open a new one when the call to TF succeeds - something like that, so we don't have ext calls inside of a transaction.

Reply:
The new tenant fetcher API is constructed in a way that it first checks whether the tenant exists in the database and it won't fetch it if it does. So this check is done there.